### ISSUE TITLE
Impossible to start a debugging session since version 1.19.4.

### ISSUE THREAD

**Issue by Adrien4193** [microsoft/vscode-cpptools#12032]
### Environment

- OS and version: Ubuntu 22.04
- VS Code: 1.86.2
- C/C++ extension: 1.19.4
- OS and version of remote machine (if applicable): not remote
- GDB / LLDB version: GDB 12.1

### Bug Summary and Steps to Reproduce

Bug Summary:

When I hit F5 on a simple C++ file, the debug panel ![image](https://github.com/microsoft/vscode-cpptools/assets/39053578/a46f9034-0b27-4245-b7b5-1b7ea115c460) appears during one second and then disappear and the session doesn't start.

Steps to reproduce:

In a new folder, create a simple C++ file:

```
#include <iostream>

int main()
{
    std::cout << "Test\n";
}
```

Press F5 and select build and debug active file.

The debug session exits immediately.

### Debugger Configurations

```shell
Automatically created by VSCode tasks.json:

{
    "tasks": [
        {
            "type": "cppbuild",
            "label": "C/C++: g++ build active file",
            "command": "/usr/bin/g++",
            "args": [
                "-fdiagnostics-color=always",
                "-g",
                "${file}",
                "-o",
                "${fileDirname}/${fileBasenameNoExtension}"
            ],
            "options": {
                "cwd": "${fileDirname}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "detail": "Task generated by Debugger."
        }
    ],
    "version": "2.0.0"
}
```

### Debugger Logs

```shell
It doesn't seem the debugger has been called.
```

### Additional Information

Downgrading the extension to version 1.18.5 solves the issue.

**Comment by browntarik** [comment]
Are you setting any breakpoints to debug the program? I know this main seem trivial but if you start a debug session without breakpoint, you will not be able to step into your code.

**Comment by Adrien4193** [comment]
Hi thanks for your reply,

Yes I put a breakpoint, sorry I should have mentioned that.

To be more precise, the program is not even started (no new terminal spawned and nothing printed to the console).

This example is just the minimal setup to reproduce, I noticed the issue while working on a bigger application which is a web service with an infinite loop (I work on it every day and I almost always start it with the debugger) so it shouldn't exit immediately.

And as I said, I don't have the issue anymore if I downgrade to v1.18.5.

Let me know if I forgot to mention something else.

**Comment by browntarik** [comment]
Could you provide the exact files you are using so I would be able to reproduce the issue using F5? This will help us easily diagnose the problem.

**Comment by Adrien4193** [comment]
@browntarik Unfortunately this C++ file is the only one I was using 😢.

I noticed the issue while working on a bigger project that I debug every day, so I tried to find a minimal example to reproduce. I just created a new empty folder with this C++ file, then press F5 and same issue as for the big one, no debugging sessions are started.

@Colengms Yes it is the same behavior with or without a launch.json and wether I click on "Debug C/C++ File" or hit F5.

As I am the last C++ dev in the team, I struggled a bit to find someone else to test on the same Ubuntu version but I finally found two colleagues to do so and they don't reproduce either. So, it seems you are right, it is something with my Ubuntu PC.

I have a Windows PC at home and I didn't reproduce the issue on it (no problems with remote-ssh to my Linux machine either).

I tried reinstalling the extension and even VSCode and I still get the issue each time I switch from 1.18.5 to 1.19.4...

If nobody else has this issue and it is specific to my environment, it is fine for me if I keep using the workaround of the version downgrade so I don't bother you more with it.

I will reinstall everything on my office PC once I have more time so we can close this ticket for now and reopen if someone else encounters the same problem ?

Thanks for the investigation !
